components/brave_wallet_ui/common/hooks/balance.ts

```diff
@@ -17,11 +18,20 @@ export default function useBalance (network: BraveWallet.NetworkInfo) {
       return ''
     }
 
+    const tokensCoinType = getTokensCoinType(networks, token)
     // Return native asset balance
-    if (token.symbol.toLowerCase() === network.symbol.toLowerCase()) {
-      return account.balance
+    if (
+      (
+        token.contractAddress === '' ||
+        // 0xeee... is used as a contractAddress
+        // of native assets in our Swap Widget and used for the 0x API.
+        token.contractAddress === '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+      ) &&
+      account.coin === tokensCoinType
```

Comment:
Can you please explain why we need to check `account.coin === tokensCoinType`?

Reply:
Left a comment in the code,
It't to distinguish between `Native Assets` that share the same `chainId` of `LocalHost` networks.